src/json_spirit/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
 add_library(json_spirit STATIC
   json_spirit_reader.cpp
   json_spirit_writer.cpp)
-target_link_libraries(json_spirit common_utf8)
+target_link_libraries(json_spirit common_utf8 common)
```

Comment:
We're linking with the static `libceph-common.a` here to prevent circular dependency with shared `libceph-common.so` (as it depends on `json_spirit`). However, do we really need `json_spirit` as a separate library overall? Is there any good reason to not embed `json_spirit_{reader,writer}.cpp` into common?

Reply:
@rzarzynski the idea is to have better modularity. `json_spirit` lives in its own directory. IMHO, it's better to compile it as a separated library from the perspective of maintainability.